test-lit/semantics/canondo02.f90

```diff
@@ -1,6 +1,6 @@
 ! negative test -- invalid labels, out of range
-
-! RUN: ${F18} -funparse-with-symbols %s 2>&1 | ${FileCheck} %s
+!RUN: %test_generic %s %flang
+! EXEC: ${F18} -funparse-with-symbols %s 2>&1 | ${FileCheck} %s
```

Comment:
I think it would make sense to use the real FileCheck. You might even be able to eliminate test_any.sh and have these tests use just FileCheck. I'm not very familiar with these tests. @schweitzpgi might have more insight.

Reply:
The intent was that test_any.sh was to be a throwaway stop gap script until such time as someone decided to replace ctest and the homegrown scripts with lit and FileCheck.

It appears that time has arrived.

Reply:
I'm happy to use lit and FileCheck for generic/any tests. Although, it makes the patch name confusing...

As for @RichBarton-Arm's comment, test_any needs %flang/%f18 to stop common.sh from complaining, but as you point out, it isn't actually used in test_any.sh. But that is yet another reason to just use actual FileCheck instead in this instance.

Reply:
This PR is about enabling lit in the test infrastructure.  A future PR can focus on test content.

Reply:
RE: my original comment - I see now that common.sh sets up the $F18 variable from the input, so that was the missing link I could not see in the original review. Apologies again.

So given my misunderstanding is corrected then we can proceed with the current plan to port to lit only. It is a bit odd that lit is calling test_any.sh which is going to try and ape lit itself and call a FileCheck-like utility but we can swiftly fix that after this PR lands. Agree to stay the course.